Intégrer complètement la modale d'import d'un fichier espèces .ods dans la page saisie des espèces

clemencefrz opened the pull request:
1. Intégrer le bon texte conformément à la maquette
2. Rajouter un bouton pour lancer l'action de Pré-remplir
3. Rajouter des warnings pour rendre compte de l'absence de fichier ou d'un fichier mal formatté

J'ai rajouté deux console.warn
- Un pour signaler qu'aucun fichier n'a été téléchargé
- Un pour signaler que le format du fichier importé semble incorrect

L'intention derrière ces consoles warn c'est de les "transformer" plus tard en message visible pour les utilisateurices


----
Edit 20/10

Suite à discussion avec David et Maïtané, Ajout d'un message d'erreur au moment d'appuyer sur Pré-remplir lorsque : 
- pas de fichier téléchargé
- fichier espèce au format incorrect (format incorrect = ods mais avec les mauvais noms de colonnes)

clemencefrz committed:
Implémenter complètement la modale d'import
- Intégrer le bon texte conformément à la maquette
- Rajouter un bouton pour lancer l'action de Pré-remplir
- Rajouter des warnings pour rendre compte de l'absence de fichier ou d'un fichier mal formatté

clemencefrz committed:
Renommer la variable file pour qu'elle soit plus explicite en fichierEspècesOds

DavidBruant reviewed (commented):
C'est plutôt chouette dans l'ensemble, merci !

clemencefrz committed:
Remplacer nomenclature par format dans le texte explicatif

clemencefrz commented on scripts/front-end/components/screens/SaisieEspèces.svelte:
```diff
@@ -292,18 +306,41 @@
+                                                        un document déjà généré avec cet outil
+                                                    </li>
+                                                    <li>
+                                                        un document .ods qui respecte la nomenclature décrite <a href="https://betagouv.github.io/pitchou/projet-pitchou/technique/fichier-especes-ods" target="_blank" rel="noopener external" title="Lien vers la page de nomenclature - nouvelle fenêtre" class="fr-link">sur cette page</a>
```
Je suis d'accord j'ai modifié

clemencefrz committed:
Jeter une erreur si le fichier des espèces n'a pas le bon format

clemencefrz commented on scripts/commun/outils-espèces.js:
```diff
@@ -319,7 +319,19 @@ async function importDescriptionMenacesEspècesFromOdsArrayBuffer_version_1(odsF
+        !(lignesFauneNonOiseauOds && lignesFauneNonOiseauOds.length >= 1) &&
+        !(lignesFloreOds && lignesFloreOds.length >= 1)
+    ) {
+        console.warn('Le fichier espèces .ods semble ne contenir aucune feuille oiseau, faune non-oiseau ou flore.')
```
> Ça me donne envie d'avoir une discussion avec Maïtané sur la gestion de ces erreurs (pas bloquante pour cette PR)

Je mets un message sur le journal pour en parler à Maïtané

https://mattermost.incubateur.net/betagouv/pl/fd7af39i978g3qwsnfw5ycs7xw

clemencefrz committed:
Quelques retours rapides et faciles d'Interface Utilisateur by Maïtané

clemencefrz committed:
Gestion des erreurs suite aux retours de Maïtané
- On utilise le style pour les erreurs dans le DSFR qui concerne le composant Ajout d'un fichier
- On affiche l'erreur au moment de cliquer sur pré-remplir

hannaeko reviewed (commented):
Très cool de voir tout ça prendre forme! Je te fais confiance sur le code, j'ai juste quelque commentaires sur l'accessibilité.

J'ai aussi une remarque sur la gestion du focus de la modale. Je sais pas si ça a ça place ici ou si c'est mieux d'en faire ici une issue. Lorsque le drop down pour pré-remplir se replie je pense que le focus va sur le bouton qui a trigger la modal quand elle se ferme mais il peut ne plus être visible donc on se retrouve avec le focus un peu nul part, peut être que ce serait mieux que le focus revienne sur le bouton initiale "Pré-remplir" si le drop down est replié?

clemencefrz committed:
Affiner le message d'erreur quand le format n'est pas correct

clemencefrz committed:
Accessibilité : améliorer le title et l'intitulé du lien

clemencefrz commented on scripts/front-end/components/screens/SaisieEspèces.svelte:
```diff
@@ -84,17 +91,40 @@
 
-        if(file){
-            const descriptionMenacesEspèces = await file.arrayBuffer()
+    async function onClickPréRemplirAvecDocumentOds(){
```
C'est une très bonne idée :-) je fais la modif

clemencefrz committed:
Accessibilité : mettre le focus sur l'input du fichier quand il y a une erreur

clemencefrz commented on scripts/front-end/components/screens/SaisieEspèces.svelte:
```diff
@@ -292,18 +322,48 @@
+                                                        un document déjà généré avec cet outil
+                                                    </li>
+                                                    <li>
+                                                        un document .ods qui respecte le format décrit <a href="https://betagouv.github.io/pitchou/projet-pitchou/technique/fichier-especes-ods" target="_blank" rel="noopener external" title="Lien vers la page qui renseigne sur le format d'un fichier espèces - nouvelle fenêtre" class="fr-link fr-text--sm">sur cette page</a>
```
j'avais fait la modif avant de voir le com, suite à ton précédent commentaire donc on est en phase :-)

hannaeko commented on scripts/front-end/components/screens/SaisieEspèces.svelte:
```diff
@@ -84,17 +91,40 @@
+            messageErreurPréRemplirAvecDocumentOds = "Une erreur est survenue au moment de cliquer sur le bouton Pré-remplir.";
+            if (erreur instanceof Error) {
+                if (erreur.cause === 'format incorrect') {
+                    messageErreurPréRemplirAvecDocumentOds = 'Le fichier ne respecte pas le document .ods qui respecte le format décrit <a href="https://betagouv.github.io/pitchou/projet-pitchou/technique/fichier-especes-ods" target="_blank" rel="noopener external" title="Lien vers la page qui renseigne sur le format d\'un fichier espèces - nouvelle fenêtre">sur cette page.</a>'
```
Merci :)

clemencefrz committed:
Mettre messageErreurPréRemplirAvecDocumentOds = undefined au bon endroit

clemencefrz commented on scripts/front-end/components/screens/SaisieEspèces.svelte:
```diff
@@ -84,17 +91,40 @@
     async function onFileInput(e){
```
Ah oui d'accord !  J'ai changé, merci :-)

clemencefrz commented on scripts/front-end/components/screens/SaisieEspèces.svelte:
```diff
@@ -84,17 +88,27 @@
+                const modale = document.querySelector('#modale-préremplir-depuis-import');
+                if (modale) {
+                    //@ts-ignore
+                    window.dsfr(modale).modal.conceal();
```
Vu avec toi : 
L'idée d'un framework c'est d'arrêter de manipuler le DOM à la place du framework qui manipule le DOM

clemencefrz committed:
Utiliser bind:this à la place de query selector

DavidBruant reviewed (approved):
Vu en synchro
